pkg/config/expand.go

```diff
@@ -503,6 +512,28 @@ func isSimpleVariable(str string) bool {
 	return matched
 }
 
+// isEscapeVariable checks if the entire string is an escaped variable \$(not.var)
+func isEscapeVariable(str string) bool {
+	// if string has multiple variables all of them must have escape character
+	if hasNoEscapeVariable(str) {
```

Comment:
I'm not sure I understand the reason for this. Is it not possible to work around existing variables? I want to make sure it's eventually possible to support the following case when expandVariable is implemented:
`\$(ls -l $(vars.list_path))`
In this case, the outer "variable" is escaped to let this run as a subshell command, but access the vars.list_path variable inside of that command.

Reply:
I added this `hasNoEscapeVariable` function to avoid this situation meanwhile we do not have expandVariable implemented but I think I got your point. Should be better call expandVariable function explicitly in such situations, is that what you thought?